src/utils/auth/api.ts

```diff
@@ -63,3 +89,39 @@ export const addReadingSession = async (chapterNumber: number, verseNumber: numb
     chapterNumber,
     verseNumber,
   });
+
+export const getUserPreferences = async (locale: string): Promise<UserPreferencesResponse> => {
+  const userPreferences = (await privateFetcher(
+    makeUserPreferencesUrl(),
+  )) as UserPreferencesResponse;
+  // if the audio Preferences are saved in the DB
+  if (userPreferences[PreferenceGroup.AUDIO]) {
+    const { reciter: reciterId } = userPreferences[PreferenceGroup.AUDIO];
+    if (reciterId) {
+      // we need to convert the id into reciter data
+      const recitersResponse = await getAvailableReciters(locale);
+      const selectedReciters = recitersResponse.reciters.filter(
+        (reciter) => reciter.id === Number(reciterId),
+      );
+      if (selectedReciters.length) {
+        const [selectedReciter] = selectedReciters;
+        userPreferences[PreferenceGroup.AUDIO] = {
+          ...userPreferences[PreferenceGroup.AUDIO],
+          reciter: selectedReciter,
+        };
+      }
+    }
+  }
+  return userPreferences;
+};
+
+export const addOrUpdateUserPreference = async (value: any, group: PreferenceGroup) =>
+  postRequest(makeUserPreferencesUrl(), {
+    value,
+    group,
+  });
+
+export const deleteUserPreference = async (group: PreferenceGroup) =>
+  deleteRequest(makeUserPreferencesUrl(), {
```

Comment:
nit: can `group` value be inside the url itself?. so `makeUserPreferencesUrl({group})`

I lef the comment on the BE PR as well.